### ISSUE TITLE
train_softmax.py loss saturated at 6

### ISSUE THREAD

**Issue by taureanamir** [davidsandberg/facenet#996]
@davidsandberg 

I am fine tuning facenet on my dataset. But I am running into case where my loss does not go below 6. Loss starts at 10 and reaches to 6 and saturates there forever.

My dataset contains over **8000 images of 27 identities** for training.

Following are the parameters that I used.
```
python3 /src/train_softmax.py \
--logs_base_dir /logs/softmax/ \
--models_base_dir /models/softmax/ \
--gpu_memory_fraction 0.45 \
--pretrained_model  /pretrained-models/20180402-114759/model-20180402-114759.ckpt-275 \
--data_dir /data/AVL-faces-final/aligned-images/train \
--model_def models.inception_resnet_v1 \
--max_nrof_epochs 500 \
--batch_size 45 \
--image_size 160 \
--epoch_size 1000 \
--embedding_size 512 \
--optimizer ADAGRAD \
--learning_rate -1 \
--keep_probability 0.8 \
--learning_rate_schedule_file /data/learning_rate_retrain_softmax.txt \
--weight_decay 5e-4 \
--validation_set_split_ratio 0.2 \
--validate_every_n_epochs 5 \
```

I **changed** the below **code** in **train_softmax.py** file.
```

# Build a Graph that trains the model with one batch of examples and updates the model parameters

    train_op = facenet.train(total_loss, global_step, args.optimizer,
         learning_rate, args.moving_average_decay, tf.global_variables(), args.log_histograms)

--> Relpaced above code with
# fine_turn ftune_vlist variables
    all_vars = tf.trainable_variables()
    ftune_vlist = [v for v in all_vars if v.name.startswith('InceptionResnetV1/Block8')]
    train_op = facenet.train(total_loss, global_step, args.optimizer,
                             learning_rate, args.moving_average_decay, ftune_vlist, args.log_histograms)

# # Create a saver
# saver = tf.train.Saver(tf.trainable_variables(), max_to_keep=3)

# Create a saver for after fine tune
    saver = tf.train.Saver(ftune_vlist, max_to_keep=3)
```
Your help will be much appreciated.

Best,
Amir.
